components/script/body.rs

```diff
@@ -127,10 +130,17 @@ impl TransmitBodyConnectHandler {
     /// in favor of making `re_extract` actually re-extract a stream from the source.
     /// See #26686
     fn transmit_source(&self) {
+        // Reset `in_memory_done` because this means there are
+        // incoming streams and this is not done yet.
+        self.in_memory_done.set(false);
```

Comment:
We need to do this as part of https://github.com/servo/servo/blob/1d0c068ae6ec0ca33965976c78b271352861b758/components/script/body.rs#L104

because if we do it here then it will reset each time and will never be `true`.

On the other hand, here we need to call `stop_reading` and return early if it is already `true`.

Reply:
ahh, yes! let me fix it! thanks! (but interestingly, current patch doesn't break any tests... 🤔)